File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/java/exercises/practice/resistor-color-trio/src/main/java/ResistorColorTrio.java
class ResistorColorTrio {
    String label(String[] colors) {
        int first = colorValue(colors[0]);
        int second = colorValue(colors[1]);
        int exponent = colorValue(colors[2]);
        
        long mainValue = first * 10L + second;
        long totalOhms = mainValue * (long) Math.pow(10, exponent);
        
        if (totalOhms >= 1_000_000_000L && totalOhms % 1_000_000_000L == 0) {
            return String.format("%d gigaohms", totalOhms / 1_000_000_000L);
        } else if (totalOhms >= 1_000_000L && totalOhms % 1_000_000L == 0) {
            return String.format("%d megaohms", totalOhms / 1_000_000L);
        } else if (totalOhms >= 1000L && totalOhms % 1000L == 0) {
            return String.format("%d kiloohms", totalOhms / 1000L);
        }
        return String.format("%d ohms", totalOhms);
    }
    
    private int colorValue(String color) {
        switch (color.toLowerCase()) {
            case "black": return 0;
            case "brown": return 1;
            case "red": return 2;
            case "orange": return 3;
            case "yellow": return 4;
            case "green": return 5;
            case "blue": return 6;
            case "violet": return 7;
            case "grey": return 8;
            case "white": return 9;
            default: throw new IllegalArgumentException("Invalid color: " + color);
        }
    }
}

File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/javascript/exercises/practice/resistor-color-trio/resistor-color-trio.js
//
// This is only a SKELETON file for the 'Resistor Color Trio' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class ResistorColorTrio {
  static COLOR_VALUES = {
    black: 0,
    brown: 1,
    red: 2,
    orange: 3,
    yellow: 4,
    green: 5,
    blue: 6,
    violet: 7,
    grey: 8,
    white: 9,
  };

  constructor(first, second, third) {
    [first, second, third].forEach(color => {
      if (!Object.hasOwnProperty.call(ResistorColorTrio.COLOR_VALUES, color)) {
        throw new Error('invalid color');
      }
    });
    this.first = first;
    this.second = second;
    this.third = third;
  }

  label() {
    const colorValues = ResistorColorTrio.COLOR_VALUES;
    const firstVal = colorValues[this.first];
    const secondVal = colorValues[this.second];
    const exponent = colorValues[this.third];
    
    let value = (firstVal * 10 + secondVal) * (10 ** exponent);
    let divisor = 1;
    let prefix = '';
    
    if (value >= 1e9 && value % 1e9 === 0) {
      divisor = 1e9;
      prefix = 'giga';
    } else if (value >= 1e6 && value % 1e6 === 0) {
      divisor = 1e6;
      prefix = 'mega';
    } else if (value >= 1e3 && value % 1e3 === 0) {
      divisor = 1e3;
      prefix = 'kilo';
    }
    
    value = value / divisor;
    
    // Handle plural for ohms when there's no prefix
    const unit = prefix ? `${prefix}ohms` : 'ohms';
    return `${value} ${unit}`;
  }
}
